dashboard/15-final/app/api/auth/[...nextauth]/route.ts

```diff
@@ -0,0 +1,53 @@
+import NextAuth, { NextAuthOptions } from 'next-auth';
+import CredentialsProvider from 'next-auth/providers/credentials';
+import bcrypt from 'bcrypt';
+import { fetchUser } from '../../../lib/data';
+import { User } from '@/app/lib/definitions';
+
+const authOptions: NextAuthOptions = {
+  providers: [
+    CredentialsProvider({
+      name: 'credentials',
+      credentials: {},
+
+      // TODO: Not sure how to type authorize function correctly
+      // @ts-ignore
+      async authorize(credentials: {
+        email: string;
+        password: string;
+      }): Promise<User | null | undefined> {
+        const { email, password } = credentials;
```

Comment:
```suggestion
      async authorize(credentials) {
        const { email, password } = credentials ?? {};
```

This is a network boundary, and the incoming object here is not verified. It would need a basic validation so it's not malicious.

Types are inferred from `credentials: {}` otherwise